calcom/cal.com
# feat: Add 'Disable on Prefill' option for fields

/claim #15712 

## What does this PR do?

Added a new checkbox to disable the input. this change provides the user the option to lock certain identifiers and if the URL parameter exists, disable the input: 

- Fixes #15712  (GitHub issue number)
- Fixes CAL-4007 (Linear issue number - should be visible at the bottom of the GitHub issue description)


https://www.loom.com/share/b09fb51e46194115b89c81f0431cb500

Demo Loom with all fields https://www.loom.com/share/63f34b4506564efeb124abcadb520489

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



- Are there environment variables that should be set?
N/A
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
1. create a new event or take any event of your choice.
2. click on the event
3. go to advanced
4. In question section. add a new question.(identifier: testprefill)
5. The question form should have a check box with the description (Disable input if the URL identifier is prefilled)
6. check the checkbox as mentioned earlier.
7. fill in other details and save.
8. update the event by clicking the save button at the top right corner.
9. Do it for any existing question(for example email)
10. Now click on Preview at the top right side.
11. fill in the month and time and submit the form
12. now edit the URL by adding new search params(&testprefill=test&email=test@gmail.com) and enter
13. The form should be replaced with the URL search parameters.
14. Email and test refill input should be disabled.
15. click on the submit button.

- Any other important info that could help to test that PR

## Checklist



- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings
